### ISSUE TITLE
Lazysizes integration together with slick slider

### ISSUE THREAD

**Issue by estefanova** [aFarkas/lazysizes#510]
Hi, there!

Recently I've implemented your plugin into my project and it works really great! I managed to enable it for content of type: _images, background_images_ and _iframe videos_. In my project I use a lot of sliders as each sliding element contains various content of the types mentioned above. My problem is that right now all the sliding items are all '_lazyloaded_' at once, immediately right after the particular slider gets visible into the view port. This is Ok to a certain extent, but as my big struggle is to reduce the page response and load time, I want to gain control over state of sliding elements and lazyload only those of them which are visible into the current active slide. I know that slick slider has lazyloading function as well, but the main reason that I choose your plugin is that it is SEO friendly, while most of the other lazyloading plugins do not guarantee that. So my question is: "Is it possible to lazyload only elements visible into the current active slide inside slick slider?" I believe that I can reduce the number of request this way and achieve some performance improvement. And if this is possible, then google will still be able to see and index all the content inside my sliders, right?

These are my lazysizes configurations which I include into the head part of my pages:
```
<script>(function(){
window.lazySizesConfig=window.lazySizesConfig||{};
window.lazySizesConfig.loadMode=1;
window.lazySizesConfig.expand=10;
window.lazySizesConfig.expFactor=1.5
})();
</script>

<script src="/plugins/ls.unveilhooks.min.js" type="text/javascript"></script>
<script src="lazysizes.min.js"></script>
```

Right after these, I include slick slider js scripts and initialization.

Looking forward for your answer! Any help or advice will be appreciated!

Greetings,

Eli

**Comment by aFarkas** [comment]
I would need an example. Please create a jsfiddle/codepen...

**Comment by a-atalla** [comment]
also if the slick slider is the first section of the page, will all the images from all the slides will be loaded immediately 
here is an example so we can check
https://codepen.io/aatalla/pen/yEExKO

**Comment by aFarkas** [comment]
@a-atalla 

I can not fully replicate your issue, but I see some problems. First what I see:

<img width="769" alt="bildschirmfoto 2018-06-24 um 10 47 56" src="https://user-images.githubusercontent.com/188254/41817488-36c0e426-779c-11e8-8363-b009a36ee985.png">

As you can see there are only three images loaded and not all and lazysizes does a pretty good job here, because lazysizes starts to download only the first visible image and then after that preloads the next two images are loaded. No other lazyloader is so intelligent to postbone the preload images.

But now to your main problem:
If you remove your slick initialization code (and this might happen in your app). lazysizes will download all images. The reason for this is that you do not specify image dimensions. At the time lazysizes initializes all images are in view. Only if those images are loaded or slick slider is initialized they are not in view anymore.

You can solve this by:
a) specifying image dimensions (check documentation) and pre defining the slider (set height and overflow hidden) good solution a little bit more work
b) initializing lazysizes after slick slider (check documentation) bad solution, but easy to implement.

**Comment by estefanova** [comment]
Hi!

Thank you for your quick answer and excuse my late response! I checked your demo and I also created mine. Here it is https://jsfiddle.net/estefanova/hopn3wv7/55/ . From what I see in my demo everything is working perfectly and as expected. I read your detailed explanations so I will recheck the implementation in my local project( like order of scripts inclusion and execution ) because I assume that probably something there is not quite Ok.

Thank you very much once again!

Greetings
